lib/metrics/index.js

```diff
@@ -53,6 +53,16 @@ class Metrics {
         return this.serviceName;
     }
 
+    // T247820: Selectively disable service label based on environment variable.
+    // Intended for production use case where the service label will be set
+    // by the Prometheus server.
+    getServiceLabel() {
```

Comment:
Perhaps we should handle that in prometheus _handleStaticLabels? The users of the metrics API can still set the `service` label and as I understand the idea is to drop it right?

Reply:
It is written this way to be as flexible as possible.  For users outside of WMF production, the service label can be whatever the user wants, including the option to get the value provided by `getServiceLabel()`.  For WMF Production, this label will be set by the Prometheus Server at scrape time.

The toggle is specifically about whether or not to provide the label in heap metrics.  The service label in heap metrics was [defined a while back](https://github.com/wikimedia/operations-deployment-charts/blob/master/charts/citoid/config/prometheus-statsd.conf#L18), but developers using service-runner expressed interest in retaining the service label present by default in the metrics endpoint.

To your question, no, the intent is not to drop the service label entirely.  Users of service-runner can still set the service label to whatever they like for any metric they define.  For instances in WMF Production, this label will be externally set.